### PR TITLE
set up example for test of integration on experimental data

### PR DESCRIPTION
**Pull request by jmfranck**

### CODE REVIEW

**Review by jmfranck — COMMENTED** [review]
we just need a docstring for the new example and to make sure the documentation still builds!

**Comment by guinneak198** [comment]
okay documentation done and build done!!!

File: .gitignore [hunk 1]
```diff
@@ -17,3 +17,8 @@ doc_src/auto_api/
 doc_src/auto_examples/
 auto_figures/
 enhancement.png
+pyspecProcScripts/B12TMPS0004_1.5.8.bin
+pyspecProcScripts/bossac.exe
+pyspecProcScripts/getBoardInfo.bat
+pyspecProcScripts/uploadFirmware.bat
+"\357\200\272q"
```
Comment (jmfranck) [inline .gitignore:24]: rather than adding to gitignore, just delete these changes.

File: examples/test_integration.py [hunk 2]
```diff
@@ -1,9 +1,22 @@
+""" Acquiring the Error Associated for Experimental Data
+========================================================
+
+Takes data from a standard echo experiment with multiple scans and compares the 
+traditional method of acquiring the error, the propagated error along the inactive
+coherence transfer (CT) pathways in the integrated signal slice, and the error associated
+with the integrated signal in the active coherence pathway.
```
Comment (jmfranck) [inline examples/test_integration.py:7]: these are all methods of estimating the error of the integral.  That's not clear here.

Also, you don't describe what the "traditional method" does.

Also, "propagated error" is not a sufficient description.  You propagate the error of XXX to estimate the error associated with YYY.

File: examples/test_integration.py [hunk 3]
```diff
@@ -1,9 +1,22 @@
...
+
+Demonstrates how our method of averaging the error along the inactive CTs compares 
+against traditional methods. With our method, there is no need to take multiple scans
+as we propagate the error in the inactive CTs instead.
```
Comment (jmfranck) [inline examples/test_integration.py:11]: w/ above changes, this will be good.

**Review by jmfranck — COMMENTED** [review]
flagged a couple issues

File: .gitignore (hunk 1 above)
Comment (guinneak198) [reply .gitignore:24]: I know how to delete the files themselves.... but how do I delete the changes?

File: examples/test_integration.py (hunk 2 above)
Comment (guinneak198) [reply examples/test_integration.py:7]: okay I'm hoping the new docstring is a little clearer....

File: .gitignore (hunk 1 above)
Comment (jmfranck) [reply .gitignore:24]: Just delete the files, and then go into .gitignore, and remove these lines.

File: examples/test_integration.py (hunk 2 above)
Comment (jmfranck) [reply examples/test_integration.py]: yes, I edited slightly, but good

File: examples/test_integration.py [hunk 4]
```diff
@@ -5,19 +5,17 @@
...
+    -   Taking the standard deviation of many integrals determined by
+        integrating over the signal bandwidth of the active CT pathway.
+        (Best method when many scans are available)
     
 Demonstrates that by propagating the error of the integral in the inactive CTs we still
-get a reasonable error within the limits of traditional methods. This then shows that 
-we no longer need multiple scans to acquire the error of the integral and can do so by 
-looking at the inactive CTs.
+get a reasonable error within the limits of traditional methods.
 """
```
Comment (jmfranck) [inline examples/test_integration.py]: see if this looks good to you

File: examples/test_error.py [hunk 5]
```diff
@@ -1,5 +1,5 @@
-""" Acquiring the Error Associated for Experimental Data
-========================================================
+""" Validate Inactive CT Error
+==============================
```
Comment (jmfranck) [inline examples/test_error.py]: changed this and the name of the file (above)

File: examples/test_error.py [hunk 6]
```diff
@@ -25,6 +25,8 @@
 import numpy as np
 rcParams['image.aspect'] = 'auto' # needed for sphinx gallery
 
+# sphinx_gallery_thumbnail_number = 4
+
```
Comment (jmfranck) [inline examples/test_error.py:29]: This picks the figure for the gallery -- actually pretty important.

File: examples/test_error.py [hunk 7]
```diff
@@ -46,8 +48,6 @@
         postproc="spincore_echo_v1",
         lookup=lookup_table,
     )
-    fl.next("raw data")
-    fl.image(s)
     s.ift("t2")
```
Comment (jmfranck) [inline examples/test_error.py]: there was still way too much extra junk

File: examples/test_error.py [hunk 8]
```diff
@@ -103,7 +101,7 @@
     # }}}
 
     # {{{Normalization
-    frq_slice = integrate_limits(select_pathway(s, signal_pathway), fl=fl)
+    frq_slice = integrate_limits(select_pathway(s, signal_pathway))
```
Comment (jmfranck) [inline examples/test_error.py]: these only show the signal energy plot, which is not informative.

File: examples/test_error.py [hunk 9]
```diff
@@ -150,33 +147,32 @@
...
     # {{{ Calculating propagated error along active CT on noise slice
-    active_error = active_propagation(s, signal_pathway, indirect="nScans", fl=fl)
+    active_error = active_propagation(s, signal_pathway, indirect="nScans")
     avg_active_error = active_error.C.mean("nScans").item()
     # }}}
 
     # {{{ Plotting Errors
-    fl.next("comparison of std")
+    fl.next("comparison of std", legend=True)
     for i in range(len(s_int_lst)):
```
Comment (jmfranck) [inline examples/test_error.py]: this positions the legend to the right of the figure, and the \n below make sure everything fits in the legend box

File: pyspecProcScripts/integrate_limits.py [hunk 10]
```diff
@@ -37,9 +37,7 @@ def integrate_limits(s, axis="t2", fl=None):
         fl.plot(signal_E['sigma':(filter_width,filter_width+1e-6)],'o', human_units=False)
         fl.pop_marker()
     fwhm = filter_width
-    fl.push_marker()
     temp.mean_all_but(axis)
     # https://en.wikipedia.org/wiki/Full_width_at_half_maximum
     temp.convolve(axis, fwhm/(2*np.sqrt(np.log(4))))
-    fl.pop_marker()
```
Comment (jmfranck) [inline pyspecProcScripts/integrate_limits.py:31]: these were just extra + actually caused a bug.

Comment (jmfranck) [reply pyspecProcScripts/integrate_limits.py:31]: (b/c they weren't inside an `if fl is None` conditional)

File: examples/test_error.py (hunk 8 above)
Comment (guinneak198) [reply examples/test_error.py]: good call!

File: pyspecProcScripts/integrate_limits.py (hunk 10 above)
Comment (guinneak198) [reply pyspecProcScripts/integrate_limits.py:31]: okay so when using push_marker/pop_marker they need to be inside an if 'fl is not None:' instance?

**Review by guinneak198 — COMMENTED** [review]
looks good on my end good call on the legend and renaming the file!

File: pyspecProcScripts/integrate_limits.py (hunk 10 above)
Comment (jmfranck) [reply pyspecProcScripts/integrate_limits.py:31]: yes

**Pull request merged**